state-sync/storage-service/types/src/lib.rs

```diff
@@ -267,17 +275,64 @@ pub struct ProtocolMetadata {
 }
 
 impl ProtocolMetadata {
-    pub fn can_service(&self, _request: &StorageServiceRequest) -> bool {
-        // TODO(philiphayes): fill out
-        true
+    pub fn can_service(&self, request: &StorageServiceRequest) -> bool {
+        use StorageServiceRequest::*;
+        match request {
+            GetServerProtocolVersion
+            | GetStorageServerSummary
+            | GetNumberOfAccountsAtVersion(_) => true,
+            GetAccountStatesChunkWithProof(request) => {
+                let chunk_size = match CompleteDataRange::new(
+                    request.start_account_index,
+                    request.end_account_index,
+                ) {
+                    Ok(range) => range.len(),
+                    Err(_) => return false,
+                };
+                self.max_account_states_chunk_size >= chunk_size
+            }
+            GetEpochEndingLedgerInfos(request) => {
+                let chunk_size =
+                    match CompleteDataRange::new(request.start_epoch, request.expected_end_epoch) {
+                        Ok(range) => range.len(),
+                        Err(_) => return false,
+                    };
+                self.max_epoch_chunk_size >= chunk_size
+            }
+            GetTransactionOutputsWithProof(request) => {
+                let chunk_size =
+                    match CompleteDataRange::new(request.start_version, request.end_version) {
+                        Ok(range) => range.len(),
+                        Err(_) => return false,
+                    };
+                self.max_transaction_output_chunk_size >= chunk_size
+            }
+            GetTransactionsWithProof(request) => {
+                let chunk_size =
+                    match CompleteDataRange::new(request.start_version, request.end_version) {
+                        Ok(range) => range.len(),
+                        Err(_) => return false,
+                    };
+                self.max_transaction_chunk_size >= chunk_size
+            }
+        }
     }
 }
 
-/// A type alias for different epochs.
-pub type Epoch = u64;
+// TODO(philiphayes): default constants in diem-config?
+impl Default for ProtocolMetadata {
```

Comment:
Future nit: we'll need to read these values from the StorageServiceConfig instead of hard coding them here 😄. Similar to what `PeerStates` does.